app/src/main/java/org/tokend/template/features/signin/logic/SignInUseCase.kt

```diff
@@ -60,9 +62,9 @@ class SignInUseCase(
     private fun getWalletInfo(email: String, password: CharArray): Single<WalletInfo> {
         val networkRequest = keyServer.getWalletInfo(email, password).toSingle()
 
-        return credentialsPersistence
-                ?.takeIf { it.getSavedEmail() == email }
-                ?.loadCredentialsMaybe(password)
+        return walletInfoPersistence
```

Comment:
Add email param to the wallet info getter so the info will be returned only if password is correct and emails match